### DISCUSSION TITLE
how to calculate the distance between a group and its child?

### DISCUSSION THREAD

**Discussion by wuxingping** [fabricjs/fabric.js#8723] [General]
how to calculate the distance between a group and it's child ? as follow：
![image](https://user-images.githubusercontent.com/41353266/220348822-2b71d6c7-d575-49b7-b8a7-94f3460d4621.png)
how to calculate the vertical and horizontal distance between a group and it's child?

**Comment by ShaMan123** [comment]
object.left/top are relative (with respect to originX/Y) to group center so you need to translate those values from the object origin to group top/left
look at `getXY`

**Reply by wuxingping** [reply]
> object.left/top are relative (with respect to originX/Y) to group center so you need to translate those values from the object origin to group top/left look at `getXY`

i am a little confused,may i have more details?

**Reply by ShaMan123** [reply]
Generally, I suggest you search the methods I reference and read them in the code base, and try them out.
The issue here is relative planes/coordinate systems. An object that belongs to a group has the values of left, top relative to the group's center point. 
For a general solution (measuring an object that is nested deep in a group against that group) you should transform all points/measurements to the canvas plane.
Think of the following:
1 step for me and for you are not similar but we both call/view our step as a single step. For an ant it is the same. If you forgot yourself and suddenly became smaller/bigger and someone told you, take a step forward you would miss the exact place you would have landed on if you were not changed. We can't measure anything properly like that so we need to convert (transform) the steps to a standard unit. That is the canvas plane. And is an implementation of linear algebra.
`sendPointToPlane` is an excellent util just for this.

**Reply by ShaMan123** [reply]
well, you can use `getCoords(true)` as well

**Reply by wuxingping** [reply]
> Generally, I suggest you search the methods I reference and read them in the code base, and try them out. The issue here is relative planes/coordinate systems. An object that belongs to a group has the values of left, top relative to the group's center point. For a general solution (measuring an object that is nested deep in a group against that group) you should transform all points/measurements to the canvas plane. Think of the following: 1 step for me and for you are not similar but we both call/view our step as a single step. For an ant it is the same. If you forgot yourself and suddenly became smaller/bigger and someone told you, take a step forward you would miss the exact place you would have landed on if you were not changed. We can't measure anything properly like that so we need to convert (transform) the steps to a standard unit. That is the canvas plane. And is an implementation of linear algebra. `sendPointToPlane` is an excellent util just for this.

thank you so much,i think maybe there was a little misunderstanding,i didn't know getXY is a method can be used directly .i have been 
reading some code in v5 and I have no impression to this method of getXY, so i  thought it was just  abstract concept.whatever,i relly appraciate for your reply.and also i had more knowleage about that.

**Reply by ShaMan123** [reply]
Yes, I forgot to say that I work on v6 so my answers are all in respect to that

**Reply by wuxingping** [reply]
> `sendPointToPlane`

https://codesandbox.io/s/border-problem-of-group-forked-qj3fqu?file=/src/App.vue
![image](https://user-images.githubusercontent.com/41353266/221496377-4e3cbad1-01df-421c-aa53-fcd5256ce24a.png)

**Reply by ShaMan123** [reply]
https://codesandbox.io/s/border-problem-of-group-forked-dh06wc?file=/src/App.vue

e.e.target is not related

**Reply by wuxingping** [reply]
> https://codesandbox.io/s/border-problem-of-group-forked-dh06wc?file=/src/App.vue
> 
> e.e.target is not related

the demo works well,but my local code didn't work.

**Reply by wuxingping** [reply]
> https://codesandbox.io/s/border-problem-of-group-forked-dh06wc?file=/src/App.vue
> 
> e.e.target is not related

if set `layout`  to 'clip-path',it works well.but if set 'layout' to 'fixed',it didn't work.you can have a try.

**Reply by ShaMan123** [reply]
this is a bug no doubt
would you open an issue?

**Reply by ShaMan123** [reply]
found the bug

**Reply by ShaMan123** [reply]
#8747 try this out

**Comment by wuxingping** [comment]
hi  @ShaMan123  there is one other question i wanna to ask as follow:
<img width="608" alt="9dd2a06f6b188c0bca668c83e56f4a1" src="https://user-images.githubusercontent.com/41353266/220892558-31657475-ce08-4845-a487-2364909d8b81.png">
isContainedWithinObject method is not enough.it also needs convert by pointer relatived,but i donot know how to do exactly,and i have tried.

**Reply by ShaMan123** [reply]
`isContainedWithinObject(otherObject, true)` should work

**Reply by wuxingping** [reply]
> `isContainedWithinObject(otherObject, true)` should work

> `isContainedWithinObject(otherObject, true)` should work

you mean image.isContainedWithinObject(group2, true).i have tried,it doesn't work.and also the method getCoords(true) above previous question doesn't work as well.

**Reply by ShaMan123** [reply]
provide a repro in v6
